Makefile

```diff
@@ -1,83 +1,87 @@
-EMILY_LAMBDA_BINARY := target/lambda/emily-handler/bootstrap.zip
-EMILY_CDK_TEMPLATE := emily/cdk/cdk.out/EmilyStack.template.json
+AUTOGENERATED_SOURCES := .generated-sources
 
 # Don't use the install target here so you can rerun install without
 # Makefile complaints.
 export DATABASE_URL=postgres://user:password@localhost:5432/signer
 
+# The package flags for cargo commands.
+CARGO_FLAGS := --workspace --exclude emily-openapi-spec --exclude blocklist-openapi-gen
+
+# ##############################################################################
+# MAIN TARGETS
+# ##############################################################################
+
 install:
 	pnpm install
 
-build: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo build ${CARGO_BUILD_ARGS}
-	pnpm --recursive build
-	# TODO(719): No need to do this once rustfmt 2.0.0 ships
-	# Format generated sources:
-	cargo fmt -p emily-openapi-spec -p emily-client -p blocklist-api -p blocklist-openapi-gen
+build: blocklist-client-codegen emily-client-codegen contracts
+	cargo build --all-targets $(CARGO_FLAGS) ${CARGO_BUILD_ARGS}
 
-test: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo test -- --test-threads=1
+test:
+	cargo nextest run --lib $(CARGO_FLAGS) --no-fail-fast ${CARGO_BUILD_ARGS}
 	pnpm --recursive test
 
-test-ci: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo nextest run --no-fail-fast
-	pnpm --recursive test
+test-build:
+	cargo test build $(CARGO_FLAGS) --no-run --locked ${CARGO_BUILD_ARGS}
 
 lint:
-	cargo clippy -- -D warnings
 	cargo fmt --all -- --check
+	cargo clippy -- -D warnings
 	pnpm --recursive run lint
 
-format: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
+format:
 	cargo fmt
 
+contracts:
+	pnpm --prefix contracts run build
+
 clean:
 	cargo clean
 	pnpm --recursive clean
 
-.PHONY: install build test lint format clean
+.PHONY: install build test test-build lint format contracts clean
 
-# Integration tests.
-# ------------------------------------------------------------------------------
+# ##############################################################################
+# NEXTEST
+# ##############################################################################
+
+NEXTEST_ARCHIVE_FILE := target/nextest/nextest-archive.tar.zst
+NEXTEST_SERIAL_ARCHIVE_FILE := target/nextest/nextest-archive-serial.tar.zst
+
+# Creates nextest archives
+nextest-archive:
+	cargo nextest archive $(CARGO_FLAGS) --lib --archive-file $(NEXTEST_ARCHIVE_FILE) ${CARGO_BUILD_ARGS}
+	cargo nextest archive $(CARGO_FLAGS) --archive-file $(NEXTEST_SERIAL_ARCHIVE_FILE) --test integration ${CARGO_BUILD_ARGS}
+
+# Runs nextest archives
+nextest-archive-run:
+	cargo nextest run --no-fail-fast --retries 2 --archive-file $(NEXTEST_ARCHIVE_FILE)
+	cargo nextest run --no-fail-fast --test-threads 1 --retries 2 --archive-file $(NEXTEST_SERIAL_ARCHIVE_FILE)
```

Comment:
Don't the retries need to be removed here as well? Or maybe not since these don't get used in CI.

Reply:
Yeah, locally it should be fine to have retries, then it's highlighted in a separate yellow "FLAKY" section in the nextest report, which imo is kinda nice to know if the test flat-out fails or if it is indeed flaky.